src/pip/_internal/commands/cache.py

```diff
@@ -37,11 +37,21 @@ class CacheCommand(Command):
     usage = """
         %prog dir
         %prog info
-        %prog list [<pattern>]
+        %prog list [<pattern>] [--abspath]
         %prog remove <pattern>
         %prog purge
     """
 
+    def add_options(self):
+        # type: () -> None
+        self.cmd_opts.add_option(
+            '--abspath',
+            dest='abspath',
+            action='store_true',
+            help='List the absolute path of wheels')
+
+        self.parser.insert_option_group(0, self.cmd_opts)
```

Comment:
Does `optparse` has something like `argparse`’s [subparsers](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_subparsers)? This flag only makes sense for `list`, and it feels wrong to be able to pass it into other subcommands to be silently ignored.

Reply:
Hi @uranusjr 

I have made the changes to use `pip cache list --format=abspath`, but I wasn't able to find an option to pass the flag just to `list`, and not to other subcommands.